### ISSUE TITLE
more false positive redundant null checks

### ISSUE THREAD

**Issue by dougxc** [spotbugs/spotbugs#1791]
While #1575 fixed some false positive redundant null checks, there are still a few unhandled patterns.

Test.java:
```
import java.io.*;
import java.nio.file.*;
public class Test {
    private static void run(Path dir) {
        try (DirectoryStream<Path> dirStream = Files.newDirectoryStream(dir)) {
            for (Path file : dirStream) {
                System.out.println(file);
            }
        } catch (IOException ioe) {
            throw new RuntimeException(ioe);
        }
    }
}
```

```
> java -version
java version "11.0.13" 2021-10-19 LTS
Java(TM) SE Runtime Environment 18.9 (build 11.0.13+9-LTS-jvmci-21.3-b03)
Java HotSpot(TM) 64-Bit Server VM 18.9 (build 11.0.13+9-LTS-jvmci-21.3-b03, mixed mode)
> javac -g -target 1.8 -source 1.8 Test.java
warning: [options] bootstrap class path not set in conjunction with -source 8
1 warning
> java -jar spotbugs-4.4.2/spotbugs.jar -textui -low -maxRank 15 Test.class

M C RCN: Nullcheck of dirStream at line 6 of value previously dereferenced in Test.run(Path)  At Test.java:[line 6]
```

**Comment by KengoTODA** [comment]
Probably duplicate of #1338
